src/Model/SharingLink.php

```diff
@@ -176,8 +176,8 @@ public function setWebHtml($val)
     */
     public function getWebUrl()
     {
-        if (array_key_exists("webUrl", $this->_propDict)) {
-            return $this->_propDict["webUrl"];
+        if (array_key_exists("link", $this->_propDict) && array_key_exists("webUrl", $this->_propDict["link"])) {
+            return $this->_propDict["link"]["webUrl"];
```

Comment:
According to [the metadata](https://graph.microsoft.com/v1.0/$metadata), there is no `link` property on `sharingLink` object.

Looking at the chaining here, it looks like the code is trying to reach `webUrl` from a `permission` object. If that is the case, it can be achieved by `$permission->getLink()->getWebUrl()`.

@camerontucker, could you confirm if the assumption about the code change here is correct and whether this works for you?

Reply:
Thanks for the response. You are indeed correct!

MS Graph docs list Permission as the return https://docs.microsoft.com/en-us/graph/api/driveitem-createlink?view=graph-rest-1.0